### ISSUE TITLE
Instead of uploading a folder, it creates a file of that name and then crashes

### ISSUE THREAD

**Issue by heinebold** [wangyucode/sftp-upload-action#31]
Basically that's it. It readz all the files, starts uploading, and crashes immediately, having "uploaded" only a single bogus file named like the subfolder it was trying to upload.

**Comment by heinebold** [comment]
It happened to me in force upload mode. I looked at the code and saw, indeed, there's a difference:

Force upload always `put`s: https://github.com/wangyucode/sftp-upload-action/blob/afd74751743cd27f82e7686294968355884b7e0c/lib/deployer.js#L61

Non forced mode differentiates between files and folders: https://github.com/wangyucode/sftp-upload-action/blob/afd74751743cd27f82e7686294968355884b7e0c/lib/deployer.js#L86

**Comment by lukasberbuer** [comment]
I can confirm this issue with v2.0.2 (assets is a directory):

```
found 0 remote files
 force upload -> /index.html
 force upload -> /registerSW.js
 force upload -> /assets
/home/runner/work/_actions/wangyucode/sftp-upload-action/v2.0.2/dist/index.js:7737
    const newError = new Error(msg);
                     ^

Error: _put: EISDIR: illegal operation on a directory, read dist/assets
    at SftpClient.fmtError (/home/runner/work/_actions/wangyucode/sftp-upload-action/v2.0.2/dist/index.js:7737:22)
    at ReadStream.<anonymous> (/home/runner/work/_actions/wangyucode/sftp-upload-action/v2.0.2/dist/index.js:8349:18)
    at Object.onceWrapper (node:events:642:26)
    at ReadStream.emit (node:events:527:28)
    at emitErrorNT (node:internal/streams/destroy:[157](xxx/actions/runs/5244630412/jobs/9470916135#step:6:158):8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  code: 'EISDIR',
  custom: true
}
```

**Comment by wangyucode** [comment]
Thanks a log, confirmed this issue, and I have fixed this issue in `v2.0.3`

**Comment by wangyucode** [comment]
Thanks a lot, confirmed this issue, and I have fixed this issue in `v2.0.3`
